annotation-processor/src/main/java/net/quickwrite/localizer/processor/PluralRuleProcessor.java

```diff
@@ -144,40 +132,42 @@ private void writePluralRuleFile(final String packageName,
                         .append(tuple.type().toUpperCase())
                         .append(";\n   }\n\n");
             }
-            builder.append("   return ").append(category.getClassName()).append(".OTHER;\n});\n");
+            builder.append("   return ").append(category.getClassName()).append(".OTHER;\n})");
 
-            generator.addStaticConstructorElement(builder.toString());
-        }
+            generator.addEnumValue(builder.toString());
 
+            for (int j = 1; j < locales.length; j++) {
+                final String result = locales[j].toUpperCase() +
+                        "(" + firstLocale + ".localizationFunction)";
 
-        generator.addMethod("""
-                public static Function<PluralOperand, PluralCategory> getPluralizer(final String key) {
-                     return PLURALIZATION_MAP.get(key);
-                }
-                """);
-
+                generator.addEnumValue(result);
+            }
+        }
 
-        generator.addMethod("""
-                private static void addRule(final String[] cultures, final Function<PluralOperand, PluralCategory> rule) {
-                    for (final String culture : cultures) {
-                        PLURALIZATION_MAP.put(culture, rule);
-                    }
+        generator.addConstructor("""
+                PluralRuleChecker(final Function<PluralOperand, PluralCategory> localizationFunction) {
+                    this.localizationFunction = localizationFunction;
                 }
                 """);
 
         generator.addMethod("""
-                private static boolean isInRange(int value, int min, int max) {
-                  return min <= value && value <= max;
-                }
-                """)
+                        public PluralCategory getCategory(final PluralOperand operand) {
+                            return this.localizationFunction.apply(operand);
+                        }
+                        """)
+                .addMethod("""
+                        private static boolean isInRange(int value, int min, int max) {
+                            return min <= value && value <= max;
+                        }
+                        """)
                 .addMethod("""
                         private static boolean isInRange(long value, int min, int max) {
-                          return min <= value && value <= max;
+                            return min <= value && value <= max;
                         }
                         """)
                 .addMethod("""
                         private static boolean isInRange(double value, int min, int max) {
-                          return min <= value && value <= max;
+                            return min <= value && value <= max;
                         }
                         """);
```

Comment:
Even though this works. It does not consider the different names of the classes that are getting added in the Annotation.